.github/workflows/main.yml

```diff
@@ -1,34 +1,99 @@
-name: VSIX
+name: Main CI/CD for Zowe CICS Explorer
 on:
   push:
-    branches: [main]
-
+  pull_request:
   workflow_dispatch:
+    inputs:
+      version:
+        description: Version to be published
+        required: true
 
 jobs:
   build_vsix:
-    name: build_vsix
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+
+    timeout-minutes: 60
+
+    concurrency:
+      group: ${{ matrix.os }}-node-${{ matrix.node-version }}-ci-${{ github.ref }}
+      cancel-in-progress: true
+
+    strategy:
+      # Continue to run tests on the other systems if one fails
+      fail-fast: false
+      matrix:
+        node-version: [16.x, 18.x]
+        # order operating systems from best to worst
+        os: [windows-latest, ubuntu-latest, macos-latest]
+
+    if: (github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository) && !contains(github.event.head_commit.message, '[ci skip]')
+
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
         with:
-          node-version: 14
-          registry-url: https://registry.npmjs.org/
+          node-version: ${{ matrix.node-version }}
 
       - name: Install Dependencies
         run: npm i
 
+      - name: Build
+        run: npm run build
+
+      - name: Lint
+        run: npm run lint
+
+      # - name: Tests
+      #   run: npm run test
+
+  deploy_vsix:
```

Comment:
Should this job have a "needs" clause so it waits for the `build_vsix` job to succeed?

We may need to re-enable the workflow for further testing 😋